### ISSUE TITLE
Fix Continuation-Passing style code example

### ISSUE THREAD

**Issue by TheElderMindseeker** [pika/pika#1227]
When trying to reproduce example from Continuation-Passing Style section in Introduction to Pika, the code throws a very uncomprehensive exception:

```
Traceback (most recent call last):
  File "pika_intro.py", line 37, in <module>
    connection.ioloop.start()
  File "~/env/lib/python3.7/site-packages/pika/adapters/select_connection.py", line 543, in start
    self._poller.start()
  File "~/env/lib/python3.7/site-packages/pika/adapters/select_connection.py", line 804, in start
    self._process_timeouts()
  File "~/env/lib/python3.7/site-packages/pika/adapters/select_connection.py", line 492, in process_timeouts
    callback()
  File "~/env/lib/python3.7/site-packages/pika/diagnostic_utils.py", line 53, in log_exception_func_wrap
    return func(*args, **kwargs)
  File "~/env/lib/python3.7/site-packages/pika/adapters/utils/io_services_utils.py", line 972, in _connection_lost_notify_async
    self._protocol.connection_lost(error)
  File "~/env/lib/python3.7/site-packages/pika/adapters/base_connection.py", line 426, in _proto_connection_lost
    'Stream connection lost: {!r}'.format(error))
  File "~/env/lib/python3.7/site-packages/pika/exceptions.py", line 271, in __repr__
    self.__class__.__name__, self.args[0])
IndexError: tuple index out of range
```

I suppose the problem is in the statement `connection.channel(on_channel_open)` in the example. The first argument of `connection.channel(...)` must be a channel number or None, if one wishes to allow Pika manage channel numbers itself. In order to solve the problem I edited the statement to be `connection.channel(on_open_callback=on_channel_open)`.

Reference: https://pika.readthedocs.io/en/stable/intro.html
Section: Continuation-Passing Style

**Comment by lukebakken** [comment]
Fixed by #1216

**Comment by lukebakken** [comment]
Please use this documentation until 1.1.0 is released:

https://pika.readthedocs.io/en/latest/intro.html
